src/Symfony/Component/DependencyInjection/Compiler/ReplaceAliasByActualDefinitionPass.php

```diff
@@ -122,4 +124,13 @@ private function updateArgumentReferences(array $arguments, $currentId, $newId)
 
         return $arguments;
     }
+
+    private function updateFactoryServiceReference($factoryService, $currentId, $newId)
```

Comment:
Well, we can indeed inline this for the 2.3 fix. I have no hard feelings about it.